banderwagon/element.go

```diff
@@ -170,9 +171,15 @@ func BatchToBytesUncompressed(elements ...*Element) [][UncompressedSize]byte {
 }
 
 func (p *Element) setBytes(buf []byte, trusted bool) error {
+	if len(buf) != CompressedSize {
+		return errors.New("invalid compressed point size")
+	}
+
 	// set the buffer which is x * SignY as X
 	var x fp.Element
-	x.SetBytes(buf)
+	if err := x.SetBytesCanonical(buf); err != nil {
+		return fmt.Errorf("invalid compressed point: %s", err)
+	}
```

Comment:
Another situation found by fuzzer. if the `X` coordinate isn't normalized, `SetBytes` will do the modulus field size.
But that means the byte serialization isn't canonical, which I think should be the case.
`SetBytesCanonical` will fail if the deserialized X coordinate is bigger than the modulus.